app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminProductPageOpenByIdActionGroup.xml

```diff
@@ -8,14 +8,12 @@
 
 <actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
               xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
-    <actionGroup name="GoToProductPageViaIDActionGroup">
-        <annotations>
-            <description>Goes to the Product edit page for the provided Product ID.</description>
-        </annotations>
+    <actionGroup name="AdminProductPageOpenByIdActionGroup">
         <arguments>
             <argument name="productId" type="string"/>
         </arguments>
 
         <amOnPage url="{{AdminProductEditPage.url(productId)}}" stepKey="goToProduct"/>
+        <waitForPageLoad stepKey="waitForProductPage"/>
```

Comment:
I just checked - in mftf inside amOnPage we already have waitForPageLoad. Maybe it’s better to increase timeout for it inside MFTF instead and not execute this method twice ?
https://github.com/magento/magento2-functional-testing-framework/blob/daae38c8959dd6bdb364dfa4c6095a5872573741/src/Magento/FunctionalTestingFramework/Module/MagentoWebDriver.php

It should fix the issues not just in this case but also in others

Reply:
You are 100% right. I'm going to change the implementation of the action group to contain only opening page.